clusterloader2/pkg/measurement/common/ooms_tracker.go

```diff
@@ -0,0 +1,200 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/informer"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	clusterOomsTrackerEnabledFlagName = "clusterOomsTrackerEnabled"
+	clusterOomsTrackerName            = "ClusterOomsTracker"
+	informerTimeout                   = time.Minute
+	oomEventReason                    = "OOMKilling"
+	oomEventMsgRegexPattern           = `Kill process (\d+) \((.+)\) score \d+ or sacrifice child\nKilled process \d+ .+ total-vm:(\d+kB), anon-rss:\d+kB, file-rss:\d+kB.*`
+)
+
+func init() {
+	if err := measurement.Register(clusterOomsTrackerName, createClusterOomsTrackerMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", clusterOomsTrackerName, err)
+	}
+}
+
+func createClusterOomsTrackerMeasurement() measurement.Measurement {
+	fieldSelector := fields.Set{
+		"reason": oomEventReason,
+	}.AsSelector().String()
+	return &clusterOomsTrackerMeasurement{
+		selector: &measurementutil.ObjectSelector{
+			FieldSelector: fieldSelector,
+			Namespace:     metav1.NamespaceAll,
+		},
+		msgRegex: regexp.MustCompile(oomEventMsgRegexPattern),
+	}
+}
+
+type clusterOomsTrackerMeasurement struct {
+	// lock for synchronized access to ooms field
+	lock      sync.Mutex
+	selector  *measurementutil.ObjectSelector
+	msgRegex  *regexp.Regexp
+	isRunning bool
+	stopCh    chan struct{}
+	ooms      []oomEvent
+}
+
+type oomEvent struct {
+	Node          string    `json:"node"`
+	Process       string    `json:"process"`
+	ProcessMemory string    `json:"memory"`
+	ProcessID     int       `json:"pid"`
+	Time          time.Time `json:"time"`
+}
+
+func (m *clusterOomsTrackerMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	clusterOomsTrackerEnabled, err := util.GetBoolOrDefault(config.Params, clusterOomsTrackerEnabledFlagName, false)
+	if err != nil {
+		return nil, err
+	}
+	if !clusterOomsTrackerEnabled {
+		klog.Info("skipping tracking of OOMs in the cluster")
+		return []measurement.Summary{}, nil
+	}
+
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+
+	switch action {
+	case "start":
+		return []measurement.Summary{}, m.start(config.ClusterFramework.GetClientSets().GetClient())
+	case "gather":
+		summary, err := m.gather()
+		if err != nil {
+			return summary, err
```

Comment:
NIT: `return nil, err` is more idiomatic

Reply:
Done.